components/TOC.js

```diff
@@ -11,8 +11,14 @@ export default function TOC({
 }) {
   if (!toc || !toc.length) return null
   const minLevel = toc.reduce((mLevel, item) => (!mLevel || item.lvl < mLevel) ? item.lvl : mLevel, 0)
-  const tocItems = toc.filter(item => item.lvl <= minLevel + depth).map(item => ({ ...item, content: item.content.replace(/[\s]?\{\#[\w\d\-_]+\}$/, '').replace(/(<([^>]+)>)/gi, '') }))
-
+  const tocItems = toc.filter(item => item.lvl <= minLevel + depth).map(item => ({
+    ...item,
+    content: item.content.replace(/[\s]?\{\#[\w\d\-_]+\}$/, '').replace(/(<([^>]+)>)/gi, ''),
+    //For TOC rendering in specification files in the spec repo we have "a" tags added manually to the spec markdown document
+    //MDX takes these "a" tags and uses them to render the "id" for headers like a-namedefinitionsapplicationaapplication
+    //slugWithATag contains transformed heading name that is later used for scroll spy identification
+    slugWithATag: item.content.replace(/<|>|"|\\|\/|=/gi, '').replace(/ /gi, '-').toLowerCase()
```

Comment:
```suggestion
    slugWithATag: item.content.replace(/\W/gi, '').replace(/\s/gi, '-').toLowerCase()
```

`\W` matches all non-alphabetic and non-numeric chars
`\s` matches all white chars

Reply:
the thing is that whitespace is also a non-numeric character, so it won't work for me as first regex will hide a whitespace and I won't be able to add `-` later

Reply:
Ok, you're right, it can stay, but for the second you can use the `\s`.

Reply:
done, thanks!